pkg/kubectl/cmd/util/helpers.go

```diff
@@ -523,18 +523,24 @@ func RecordChangeCause(obj runtime.Object, changeCause string) error {
 // ChangeResourcePatch creates a strategic merge patch between the origin input resource info
 // and the annotated with change-cause input resource info.
 func ChangeResourcePatch(info *resource.Info, changeCause string) ([]byte, error) {
-	oldData, err := json.Marshal(info.Object)
+	// Get a versioned object
+	obj, err := info.Mapping.ConvertToVersion(info.Object, info.Mapping.GroupVersionKind.GroupVersion())
```

Comment:
I thought versioned object came from the user, and object came from the server...